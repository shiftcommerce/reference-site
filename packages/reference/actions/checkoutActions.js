// actionTypes
import * as actionTypes from './actionTypes'

// checkout Handler
import * as checkoutHandler from './handlers/checkoutHandler'

// localForage
import LocalForage from '../lib/localforage'
const localForage = new LocalForage()

export function inputChange (formName, fieldName, fieldValue) {
  return (dispatch) => {
    dispatch(storeInputChange(formName, fieldName, fieldValue))
  }
}
// Store the input change info in the local redux store
export function storeInputChange (formName, fieldName, fieldValue) {
  return {
    type: actionTypes.SET_CHECKOUT_INPUT_VALUE,
    payload: {
      formName: formName,
      fieldName: fieldName,
      fieldValue: fieldValue
    }
  }
}

export function showField (formName, fieldName) {
  return {
    type: actionTypes.SET_CHECKOUT_INPUT_SHOWN,
    payload: {
      formName: formName,
      fieldName: fieldName
    }
  }
}

export function toggleCollapsed (componentName) {
  return {
    type: actionTypes.TOGGLE_CHECKOUT_COMPONENT_COLLAPSED,
    payload: {
      componentName: componentName
    }
  }
}

export function inputComplete () {
  return (dispatch, getState) => {
    const checkout = getState().checkout
    storeCheckoutInLocalStorage(checkout)
  }
}

// Stores a checkout in redux store
export function storeCheckout (checkout) {
  return {
    type: actionTypes.STORE_CHECKOUT,
    checkout: checkout
  }
}

// set shipping method details
export function setShippingMethod (shippingMethod) {
  return (dispatch, getState) => {
    const checkout = checkoutHandler.updateShippingMethod(getState().checkout, shippingMethod)
    dispatch(storeShippingMethod(checkout))
    storeCheckoutInLocalStorage(checkout)
  }
}

// Store shipping method details
export function storeShippingMethod (checkout) {
  return {
    payload: checkout,
    type: actionTypes.SET_SHIPPING_METHOD
  }
}

// Stores checkout in indexedDB, eventually we can update and/or duplicate this for API?
export function storeCheckoutInLocalStorage (checkout) {
  localForage.setValue('checkout', checkout)
}

// This function will be called by checkout every time the component is mounted
export function readCheckoutFromLocalStorage () {
  return (dispatch, getState) => {
    let checkout = getState().checkout
    // check if local redux store has no data
    if (checkout.updatedAt === undefined) {
      localForage.getValue('checkout').then((checkout) => {
        // if indexedDB is also empty, initialize an empty checkout
        if (checkout === null) {
          checkout = {
            shippingAddress: {},
            billingAddress: {},
            paymentMethod: {},
            shippingAddressAsBillingAddress: true,
            shippingMethod: {},
            currentStep: 1
          }
        }
        // update local redux store
        dispatch(storeCheckout(checkout))
      })
    // if local redux store has checkout data, dispatch it
    } else {
      dispatch(storeCheckout(checkout))
    }
    return checkout
  }
}

export function setShippingBillingAddress (formName, fieldName, fieldValue) {
  return (dispatch, getState) => {
    dispatch(setBillingAsShippingKey(formName, fieldName, fieldValue))
  }
}

function setBillingAsShippingKey (formName, fieldName, fieldValue) {
  return {
    type: actionTypes.SET_SHIPPING_AS_BILLING_KEY,
    payload: {
      formName: formName,
      fieldName: fieldName,
      fieldValue: fieldValue
    }
  }
}

export function changePaymentMethod (paymentMethod) {
  return {
    type: actionTypes.CHANGE_PAYMENT_METHOD,
    paymentMethod: paymentMethod
  }
}
