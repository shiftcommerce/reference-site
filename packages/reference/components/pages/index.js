import CategorySidebar from './categorySidebar'

const PageTemplates = {
  category_sidebar_template: CategorySidebar
}

export default PageTemplates
