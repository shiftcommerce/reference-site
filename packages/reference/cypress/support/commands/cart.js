/**
 * Adds variant to cart
 */
Cypress.Commands.add('addVariantToCart', () => {
  // Setup server
  cy.server()

  // Stub add to cart request
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-13.json'
  }).as('addToCart')

  // Navigate to PDP
  cy.visit('/products/clock_computer')
  // Select the instock variant
  cy.contains(/Clock Computer 13'' - £799.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCart')
})

/**
 * Adds variants to cart
 */
Cypress.Commands.add('addVariantsToCart', () => {
  // Setup server
  cy.server()

  // Stub first add to cart requests
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-13.json'
  }).as('addToCartClockComputer13')

  // Navigate to PDP
  cy.visit('/products/clock_computer')
  // Select the first variant
  cy.contains(/Clock Computer 13'' - £799.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCartClockComputer13')
  // Click the cross to close minibag
  cy.get('.c-minibag__dropdown-cross').click()

  // Stub second add to cart requests
  cy.route({
    method: 'POST',
    url: '/addToCart',
    status: 200,
    response: 'fixture:cart/add-clock-computer-15.json'
  }).as('addToCartClockComputer15')

  // Select the second variant
  cy.contains(/Clock Computer 15'' - £999.00/i).click()
  // Add item to cart
  cy.contains(/Add To Basket/i).click()
  // Check request has been made
  cy.wait('@addToCartClockComputer15')
})

/**
 * Adds promotion code to cart
 * @param {string} promoCode - the promotion code
 */
Cypress.Commands.add('addPromotionCodeToCart', ({ promoCode }) => {
  // Check the promotion form title
  cy.get('.c-coupon-form__title').contains(/Promotion Code/i)
  // Enter a promotion code
  cy.get('.c-coupon-form__input').type(promoCode)
  // Click apply button
  cy.get('.c-coupon-form__button').click()
})
