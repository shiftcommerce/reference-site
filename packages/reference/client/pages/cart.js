// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Head from 'next/head'

// Lib
import { algoliaReduxWrapper } from '../lib/algolia-redux-wrapper'
import { suffixWithStoreName } from '../lib/suffix-with-store-name'

// Actions
import { updateQuantity } from '../actions/cart-actions'

// Components
import CartTable from '../components/cart/cart-table'
import Loading from '../components/loading'

export class CartPage extends Component {
  constructor (props) {
    super(props)

    this.updateQuantity = this.updateQuantity.bind(this)
    this.deleteItem = this.deleteItem.bind(this)
  }

  updateQuantity (e) {
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: parseInt(e.target.value, 10)
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  deleteItem (e) {
    e.preventDefault()
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: 0
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  render () {
    const { cart, cart: { loading, error } } = this.props

    if (loading) {
      return (
        <Loading />
      )
    } else if (error) {
      return (
        <p>{ error }</p>
      )
    } else {
      return (
        <>
          <Head>
            <title>{ suffixWithStoreName('Your Shopping Cart') }</title>
          </Head>
          <CartTable cart={cart} updateQuantity={this.updateQuantity} deleteItem={this.deleteItem} />
        </>
      )
    }
  }
}

function mapStateToProps (state) {
  const { cart } = state
  return { cart }
}

export default algoliaReduxWrapper(connect(mapStateToProps)(CartPage), CartPage)
