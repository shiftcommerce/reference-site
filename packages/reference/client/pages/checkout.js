// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Router from 'next/router'

// Libs
import InputFieldValidator from '../lib/input-field-validator'

// Actions
import { readCheckoutFromLocalStorage,
  toggleCollapsed,
  setShippingMethod,
  changeBillingAddress,
  setValidationMessage,
  changePaymentMethod,
  inputChange,
  inputComplete,
  showField } from '../actions/checkout-actions'
import { readCart, initializeCart, updateQuantity } from '../actions/cart-actions'
import {
  createOrder,
  requestCardToken,
  setCardToken,
  setPaymentError,
  setCardErrors
} from '../actions/order-actions'

// Objects
import Logo from '../objects/logo'

// Components
import CustomHead from '../components/custom-head'
import CheckoutSteps from '../components/checkout/checkout-steps'
import AddressForm from '../components/checkout/address-form'
import CheckoutCart from '../components/checkout/checkout-cart'
import CheckoutCartTotal from '../components/checkout/checkout-cart-total'
import ShippingMethods from '../components/checkout/shipping-methods'
import PaymentMethod from '../components/checkout/payment-method'
import PromoInput from '../components/promo-input'

export class CheckoutPage extends Component {
  static async getInitialProps ({ reduxStore }) {
    return {}
  }
  constructor () {
    super()

    this.onToggleCollapsed = this.onToggleCollapsed.bind(this)
    this.nextSection = this.nextSection.bind(this)
    this.setShippingMethod = this.setShippingMethod.bind(this)
    this.onInputChange = this.onInputChange.bind(this)
    this.onInputBlur = this.onInputBlur.bind(this)
    this.onShowField = this.onShowField.bind(this)
    this.changeBillingAddress = this.changeBillingAddress.bind(this)
    this.convertToOrder = this.convertToOrder.bind(this)
    this.onPaymentMethodChanged = this.onPaymentMethodChanged.bind(this)
    this.onCardTokenReceived = this.onCardTokenReceived.bind(this)
    this.setCardErrors = this.setCardErrors.bind(this)
    this.updateQuantity = this.updateQuantity.bind(this)
    this.deleteItem = this.deleteItem.bind(this)
  }

  componentDidMount () {
    this.props.dispatch(readCheckoutFromLocalStorage())
    this.props.dispatch(readCart()).then((cart) => {
      if (cart.totalQuantity === 0) Router.push('/cart')
    })
  }

  componentWillReceiveProps (props) {
    const prevOrder = this.props.order
    const newOrder = props.order
    // Only when new order got created,
    // Redirect to order confirmation page
    if (newOrder && newOrder.id && newOrder.id !== prevOrder.id) {
      Router.push('/order')
      props.dispatch(initializeCart())
    }
  }

  updateQuantity (e) {
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: parseInt(e.target.value, 10)
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  deleteItem (e) {
    e.preventDefault()
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: 0
    }
    this.props.dispatch(updateQuantity(lineItem))
    if (this.props.cart.lineItems.length === 0) {
      Router.push('/cart')
    }
  }

  nextSection (eventType) {
    const { checkout } = this.props

    let componentName

    if (checkout.currentStep === 1) {
      componentName = 'shippingAddress'
    } else if (checkout.currentStep === 2) {
      componentName = 'shippingMethod'
    } else if (checkout.currentStep === 3) {
      componentName = 'paymentMethod'
    }

    window.scrollTo(0, 0)
    this.props.dispatch(toggleCollapsed(eventType, componentName))
  }

  onToggleCollapsed (eventType, componentName) {
    window.scrollTo(0, 0)
    this.props.dispatch(toggleCollapsed(eventType, componentName))
  }

  setShippingMethod (shippingMethod) {
    this.props.dispatch(setShippingMethod(shippingMethod))
  }

  onInputChange (event, formName, fieldName, fieldValue) {
    this.props.dispatch(inputChange(formName, fieldName, fieldValue))
  }

  validateInput (formName, fieldName, fieldValue, rules) {
    let validationMessage = new InputFieldValidator(fieldName, fieldValue, rules).validate()
    this.props.dispatch(setValidationMessage(formName, fieldName, validationMessage))
  }

  onInputBlur (event, formName, fieldName, fieldValue, rules) {
    this.validateInput(formName, fieldName, fieldValue, rules)
    this.props.dispatch(inputComplete())
  }

  convertToOrder (event) {
    if (this.props.checkout.paymentMethod.selectedMethod === 'card') {
      this.props.dispatch(requestCardToken(true))
    } else {
      this.props.dispatch(createOrder(this.props.cart, this.props.checkout, this.props.order))
    }
  }

  onShowField (formName, fieldName) {
    this.props.dispatch(showField(formName, fieldName))
  }

  onPaymentMethodChanged (paymentMethod) {
    this.props.dispatch(changePaymentMethod(paymentMethod))
  }

  onCardTokenReceived ({ error, token }) {
    if (error) {
      this.props.dispatch(setPaymentError(error.message))
    } else {
      this.props.dispatch(setCardToken(token))
    }
    this.props.dispatch(requestCardToken(false))
  }

  setCardErrors (error) {
    this.props.dispatch(setCardErrors(error))
  }

  changeBillingAddress (event, formName, fieldName) {
    this.props.dispatch(changeBillingAddress(formName, fieldName, event.target.checked))
  }

  isPaymentValid () {
    let billingAddressErrors = this.props.checkout.billingAddress.errors
    const isCardValid = !this.props.order.cardErrors
    const isBillingAddressValid = Object.keys(billingAddressErrors).every((key) => billingAddressErrors[key] === '') === true
    return isBillingAddressValid && isCardValid
  }

  render () {
    const { checkout, cart } = this.props
    const hasLineItems = cart.totalQuantity > 0
    if (checkout.loading) {
      return (
        <div>
          <CustomHead />
          loading...
        </div>
      )
    } else if (checkout.error) {
      return (
        <p>{ checkout.error }</p>
      )
    } else {
      return (
        <>
          <CustomHead />
          {hasLineItems &&
            <>
              <div className='o-header'>
                <div className='c-step-indicators__logo'>
                  <Logo className='u-text-color--primary' />
                </div>
                <div>
                  <CheckoutSteps {...this.props} />
                </div>
              </div>
              <div className='o-grid-container'>
                <div className='o-col-1-13 o-col-1-9-l'>
                  <AddressForm {...this.props}
                    title='Shipping Address'
                    formName='shippingAddress'
                    addressType='shipping'
                    onChange={this.onInputChange}
                    onBlur={this.onInputBlur}
                    onShowField={this.onShowField}
                    onToggleCollapsed={this.onToggleCollapsed}
                  />
                  <ShippingMethods
                    {...this.props}
                    formName='shippingMethod'
                    setShippingMethod={this.setShippingMethod}
                    onToggleCollapsed={this.onToggleCollapsed}
                  />
                  <PaymentMethod
                    formName='paymentMethod'
                    {...this.props}
                    changeBillingAddress={this.changeBillingAddress}
                    onChange={this.onInputChange}
                    onBlur={this.onInputBlur}
                    onPaymentMethodChanged={this.onPaymentMethodChanged}
                    onToggleCollapsed={this.onToggleCollapsed}
                    onCardTokenReceived={this.onCardTokenReceived}
                    setCardErrors={this.setCardErrors}
                  />
                </div>
                <PromoInput />
                <div className='o-col-1-13 o-col-9-13-l c-cart-table'>
                  <CheckoutCart title='Your Cart' {...this.props} updateQuantity={this.updateQuantity} deleteItem={this.deleteItem} />
                  <CheckoutCartTotal {...this.props} convertToOrder={this.convertToOrder} onClick={() => { this.nextSection('complete') }} />
                </div>
              </div>
            </>
          }
        </>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    checkout: state.checkout || {},
    cart: state.cart || {},
    order: state.order || {},
    ip: state.ip
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: dispatch
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CheckoutPage)
