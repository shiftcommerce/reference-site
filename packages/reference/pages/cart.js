// Libraries
import { Component } from 'react'
import withRedux from 'next-redux-wrapper'

// Utils
import { configureStore } from '../utils/configureStore'

// Actions
import { updateQuantity } from '../actions/cartActions'
import { readMenu } from '../actions/menuActions'

// Components
import Layout from '../components/Layout'
import CartTable from '../components/cart/CartTable.js'

export class CartPage extends Component {
  static async getInitialProps ({ store }) {
    await store.dispatch(readMenu(store))
  }

  constructor (props) {
    super(props)

    this.updateQuantity = this.updateQuantity.bind(this)
  }

  updateQuantity (e) {
    const lineItem = {
      sku: e.target.dataset.variant,
      quantity: parseInt(e.target.value, 10)
    }
    this.props.dispatch(updateQuantity(lineItem))
  }

  render () {
    const cart = this.props.cart

    if (cart.loading) {
      return (
        <Layout>
          <p>loading...</p>
        </Layout>
      )
    } else if (cart.error) {
      return (
        <Layout>
          <p>{cart.error}</p>
        </Layout>
      )
    } else {
      return (
        <Layout>
          <CartTable cart={cart} updateQuantity={this.updateQuantity} />
        </Layout>
      )
    }
  }
}

const mapStateToProps = (state) => {
  return {
    cart: state.cart || {}
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: dispatch
  }
}

export default withRedux(configureStore, mapStateToProps, mapDispatchToProps)(CartPage)
