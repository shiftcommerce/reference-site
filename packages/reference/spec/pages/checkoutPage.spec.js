// Libraries
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'

// Pages
import { CheckoutPage } from '../../pages/checkout'

// Actions
import { setShippingMethod } from '../../actions/checkoutActions'

// Fixtures
import cart from '../fixtures/cart.fixture'
import checkout from '../fixtures/checkout.fixture'
import shippingMethods from '../fixtures/shippingMethods.fixture'

test('dispatch setShippingMethod action on changing sipping method', () => {
  // Arrange
  const initialState = {}
  const expectedFunction = setShippingMethod().toString()
  const setShippingMethodSpy = jest.spyOn(CheckoutPage.prototype, 'setShippingMethod')
  const dispatch = jest.fn().mockImplementation((updateSpy) => 'first call')
  const selectedShippingMethod = shippingMethods.shippingMethods[0]

  // Act
  const wrapper = mount(
    <Provider store={createMockStore(initialState)}>
      <CheckoutPage checkout={checkout} cart={cart} dispatch={dispatch} />
    </Provider>
  )

  // Assert
  expect(wrapper).toMatchSnapshot()

  // Verify if cart line items are available
  expect(wrapper).toIncludeText('1 item')

  // To clear the logs of dispatch being called on component mount
  dispatch.mockClear()

  // Trigger quantity change
  wrapper.find(`input[id="${selectedShippingMethod.id}"]`).simulate('change', {target: { checked: true }})

  // Verify if setShippingMethod method is getting clicking shipping method
  expect(setShippingMethodSpy).toHaveBeenCalled()

  expect(dispatch).toHaveBeenCalled()

  // Verify it dispatch method called a function
  expect(dispatch.mock.calls[0][0]).toEqual(expect.any(Function))

  // Verify if the dispatch function has dispatched updateQuantity action.
  expect(dispatch.mock.calls[0][0].toString()).toMatch(expectedFunction)
})
