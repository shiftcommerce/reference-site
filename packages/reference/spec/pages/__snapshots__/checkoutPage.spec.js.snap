// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatch setShippingMethod action on changing sipping method 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <CheckoutPage
    cart={
      Object {
        "lineItems": Array [
          Object {
            "discount": 0,
            "imageUrl": "",
            "price": 10,
            "productID": 1,
            "productSku": "1231",
            "quantity": 2,
            "size": "size - 8",
            "sku": "123",
            "title": "test",
          },
        ],
      }
    }
    checkout={
      Object {
        "shippingAddress": Object {},
        "shippingMethod": Object {},
      }
    }
    dispatch={[Function]}
  >
    <div>
      <div
        className="o-header"
      >
        <span
          className="o-header__logo"
        >
          <Link
            as="/"
            href="/home/index"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <Image
                height={80}
                width={200}
              >
                <div
                  className="c-dummy_image"
                  style={
                    Object {
                      "height": 80,
                      "width": 200,
                    }
                  }
                />
              </Image>
            </a>
          </Link>
        </span>
      </div>
      <div>
        <AddressForm
          addressType="shipping"
          cart={
            Object {
              "lineItems": Array [
                Object {
                  "discount": 0,
                  "imageUrl": "",
                  "price": 10,
                  "productID": 1,
                  "productSku": "1231",
                  "quantity": 2,
                  "size": "size - 8",
                  "sku": "123",
                  "title": "test",
                },
              ],
            }
          }
          checkout={
            Object {
              "shippingAddress": Object {},
              "shippingMethod": Object {},
            }
          }
          dispatch={[Function]}
          formName="shippingAddress"
          title="Shipping Address"
        >
          <div
            className="o-form"
          >
            <h3>
              Shipping Address
            </h3>
            <form
              className="o-form__wrapper"
            >
              <SelectInputGroup
                dispatch={[Function]}
                fieldLabel="Country"
                fieldName="country"
                formName="shippingAddress"
                options={
                  Array [
                    Object {
                      "id": 1,
                      "title": "United Kingdom",
                      "value": "GB",
                    },
                    Object {
                      "id": 2,
                      "title": "United States",
                      "value": "US",
                    },
                  ]
                }
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="country"
                  >
                    Country
                     *
                  </label>
                  <select
                    id="country"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                  >
                    <option
                      value="GB"
                    >
                      United Kingdom
                    </option>
                    <option
                      value="US"
                    >
                      United States
                    </option>
                  </select>
                </div>
              </SelectInputGroup>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="Full Name"
                fieldName="fullName"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="fullName"
                  >
                    Full Name
                     *
                  </label>
                  <input
                    id="fullName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <a
                href="#"
                onClick={[Function]}
              >
                <p>
                  + Add Company Name (optional)
                </p>
              </a>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="Address 1"
                fieldName="address1"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="address1"
                  >
                    Address 1
                     *
                  </label>
                  <input
                    id="address1"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <a
                href="#"
                onClick={[Function]}
              >
                <p>
                  + Add Address 2 (optional)
                </p>
              </a>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="Post Code"
                fieldName="postCode"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="postCode"
                  >
                    Post Code
                     *
                  </label>
                  <input
                    id="postCode"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="City"
                fieldName="city"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="city"
                  >
                    City
                     *
                  </label>
                  <input
                    id="city"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="County"
                fieldName="county"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="county"
                  >
                    County
                     *
                  </label>
                  <input
                    id="county"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="Phone"
                fieldName="phone"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="phone"
                  >
                    Phone
                     *
                  </label>
                  <input
                    id="phone"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <InputGroup
                dispatch={[Function]}
                fieldLabel="Email"
                fieldName="email"
                formName="shippingAddress"
                required="true"
              >
                <div
                  className="o-form__input-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                     *
                  </label>
                  <input
                    id="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required="true"
                    type="text"
                  />
                </div>
              </InputGroup>
              <p>
                * Denotes required fields
              </p>
              <div>
                <CheckboxInputGroup
                  dispatch={[Function]}
                  fieldLabel="Sign up for Weekly Newsletters (optional)"
                  fieldName="newsletterOptIn"
                  formName="shippingAddress"
                >
                  <div
                    className="o-form__input-group"
                  >
                    <input
                      id="newsletterOptIn"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <label
                      htmlFor="newsletterOptIn"
                    >
                      Sign up for Weekly Newsletters (optional)
                    </label>
                  </div>
                </CheckboxInputGroup>
                <div
                  className="o-form__input-group"
                >
                  <button
                    onClick={[Function]}
                    type="submit"
                  >
                    View Shipping Options
                  </button>
                </div>
              </div>
            </form>
          </div>
        </AddressForm>
        <CheckoutCart
          cart={
            Object {
              "lineItems": Array [
                Object {
                  "discount": 0,
                  "imageUrl": "",
                  "price": 10,
                  "productID": 1,
                  "productSku": "1231",
                  "quantity": 2,
                  "size": "size - 8",
                  "sku": "123",
                  "title": "test",
                },
              ],
            }
          }
          checkout={
            Object {
              "shippingAddress": Object {},
              "shippingMethod": Object {},
            }
          }
          dispatch={[Function]}
          title="Your Cart"
        >
          <div
            aria-label="Cart items summary"
            className="o-checkout-cart"
          >
            <h3>
              Your Cart
            </h3>
            <Link
              href="/cart"
            >
              <a
                aria-label="Edit your cart"
                className="c-button"
                href="/cart"
                onClick={[Function]}
              >
                Edit
              </a>
            </Link>
            <div
              className="o-checkout-cart__wrapper"
            >
              <div
                aria-label="Product"
              >
                <Image
                  aria-label="Product image"
                  className="o-checkout-cart__image"
                  height={80}
                  width={80}
                >
                  <div
                    className="c-dummy_image o-checkout-cart__image"
                    style={
                      Object {
                        "height": 80,
                        "width": 80,
                      }
                    }
                  />
                </Image>
                <ul
                  className="o-checkout-cart__info"
                >
                  <li
                    aria-label="Product title"
                  >
                    test
                  </li>
                  <li
                    aria-label="Quantity"
                  >
                    Qty: 
                    2
                  </li>
                  <li
                    aria-label="Price"
                  >
                    Price: £
                    10
                  </li>
                  <li
                    aria-label="Subtotal"
                  >
                    Subtotal: £
                    20
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </CheckoutCart>
        <CheckoutCartTotal
          cart={
            Object {
              "lineItems": Array [
                Object {
                  "discount": 0,
                  "imageUrl": "",
                  "price": 10,
                  "productID": 1,
                  "productSku": "1231",
                  "quantity": 2,
                  "size": "size - 8",
                  "sku": "123",
                  "title": "test",
                },
              ],
            }
          }
          checkout={
            Object {
              "shippingAddress": Object {},
              "shippingMethod": Object {},
            }
          }
          dispatch={[Function]}
        >
          <div
            aria-label="Cart total summary"
            className="o-checkout-cart-total"
          >
            <div
              className="o-checkout-cart-total__wrapper"
            >
              <dl
                aria-label="Subtotal"
              >
                <dt>
                   Subtotal: 
                </dt>
                <dd>
                   £
                  20
                   
                </dd>
              </dl>
              <dl
                aria-label="VAT"
              >
                <dt>
                   VAT: 
                </dt>
                <dd>
                   £
                  0
                   
                </dd>
              </dl>
              <dl
                aria-label="Shipping cost"
              >
                <dt>
                   Shipping: 
                </dt>
                <dd>
                   Enter address 
                </dd>
              </dl>
              <dl
                aria-label="Total"
                className="o-checkout-cart-total__total"
              >
                <dt>
                   You Pay: 
                </dt>
                <dd>
                   
                  <b>
                    £
                    20
                  </b>
                   
                </dd>
              </dl>
            </div>
          </div>
        </CheckoutCartTotal>
        <ShippingMethods
          cart={
            Object {
              "lineItems": Array [
                Object {
                  "discount": 0,
                  "imageUrl": "",
                  "price": 10,
                  "productID": 1,
                  "productSku": "1231",
                  "quantity": 2,
                  "size": "size - 8",
                  "sku": "123",
                  "title": "test",
                },
              ],
            }
          }
          checkout={
            Object {
              "shippingAddress": Object {},
              "shippingMethod": Object {},
            }
          }
          dispatch={[Function]}
          setShippingMethod={[Function]}
        >
          <div
            aria-label="Shipping Methods"
            className="o-form"
          >
            <h3>
              Your Shipping Method
            </h3>
            <form
              className="o-form__wrapper c-shipping-method-list"
            >
              <p>
                📍 Shipping from ...
                <span
                  className="c-shipping-method-list__item-count"
                >
                  1 item
                </span>
              </p>
              <div
                aria-label="super_saver"
                className="o-form__input-group c-shipping-method-list__input-group"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  id="1"
                  name="shipping_method"
                  onChange={[Function]}
                  type="radio"
                  value="super_saver"
                />
                <label
                  htmlFor="super_saver"
                >
                  <span
                    className="c-shipping-method-list__cost"
                  >
                    £
                    3.45
                  </span>
                  <span
                    className="c-shipping-method-list__title"
                  >
                    Super Saver
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date-label"
                  >
                    Estimated Delivery: 
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date"
                  >
                    Monday 3rd December
                  </span>
                </label>
              </div>
              <div
                aria-label="ups_next_day"
                className="o-form__input-group c-shipping-method-list__input-group"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  id="2"
                  name="shipping_method"
                  onChange={[Function]}
                  type="radio"
                  value="ups_next_day"
                />
                <label
                  htmlFor="ups_next_day"
                >
                  <span
                    className="c-shipping-method-list__cost"
                  >
                    £
                    5.45
                  </span>
                  <span
                    className="c-shipping-method-list__title"
                  >
                    UPS Next Day
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date-label"
                  >
                    Estimated Delivery: 
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date"
                  >
                    Sunday 2nd December
                  </span>
                </label>
              </div>
              <div
                aria-label="ups_next_day_before_noon"
                className="o-form__input-group c-shipping-method-list__input-group"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  id="3"
                  name="shipping_method"
                  onChange={[Function]}
                  type="radio"
                  value="ups_next_day_before_noon"
                />
                <label
                  htmlFor="ups_next_day_before_noon"
                >
                  <span
                    className="c-shipping-method-list__cost"
                  >
                    £
                    8.45
                  </span>
                  <span
                    className="c-shipping-method-list__title"
                  >
                    UPS Next Day Before Noon
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date-label"
                  >
                    Estimated Delivery: 
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date"
                  >
                    Sunday 2nd December
                  </span>
                </label>
              </div>
              <div
                aria-label="ups_same_day"
                className="o-form__input-group c-shipping-method-list__input-group"
                onClick={[Function]}
              >
                <input
                  checked={false}
                  id="4"
                  name="shipping_method"
                  onChange={[Function]}
                  type="radio"
                  value="ups_same_day"
                />
                <label
                  htmlFor="ups_same_day"
                >
                  <span
                    className="c-shipping-method-list__cost"
                  >
                    £
                    10.45
                  </span>
                  <span
                    className="c-shipping-method-list__title"
                  >
                    UPS Same Day
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date-label"
                  >
                    Estimated Delivery: 
                  </span>
                  <span
                    className="c-shipping-method-list__delivery-date"
                  >
                    Today
                  </span>
                </label>
              </div>
              <button
                type="submit"
              >
                Continue to Payment
              </button>
            </form>
          </div>
        </ShippingMethods>
      </div>
    </div>
  </CheckoutPage>
</Provider>
`;
